pandas/core/methods/selectn.py

```diff
@@ -139,7 +139,10 @@ def compute(self, method: str) -> Series:
 
         # arr passed into kth_smallest must be contiguous. We copy
         # here because kth_smallest will modify its input
-        kth_val = libalgos.kth_smallest(arr.copy(order="C"), n - 1)
+        if n == 0:  # avoid OOB access with kth_smallest_c
+            kth_val = 0
+        else:
+            kth_val = libalgos.kth_smallest(arr.copy(order="C"), n - 1)
```

Comment:
```suggestion
        # avoid OOB access with kth_smallest_c
        kth_val = libalgos.kth_smallest(arr.copy(order="C"), max(n - 1, 0))
```

I would be more comfortable with this change. Since `kth_val` is a value in the array and comparisons are made with this value, this may be incorrect if values in `arr` were all negative for example

Reply:
Nice idea